src/model.jl

```diff
@@ -8,90 +9,180 @@ mutable struct Model
-
+  
     radiative_cooling_co2 = calc_radiative_cooling_co2(co2_concentration)
     
     # Read parameters
     
     geography = read_geography(joinpath(@__DIR__, "..", "input", "The_World.dat"),nx,ny)
+    
+    compute_sea_ice_extent = compute_sea_ice_extent
+    year = sea_ice_extent_year
 
-  
     compute_albedo = compute_albedo
     if compute_albedo == false
       albedo = read_albedo(joinpath(@__DIR__, "..", "input", "albedo.dat"),nx,ny)
     else
-
+  
       albedo = calc_albedo(geography,nx,ny)
     end 
 
 
+  
     diffusion_coeff = calc_diffusion_coefficients(geography,nx,ny)
     heat_capacity, tau_land, tau_snow, tau_sea_ice, tau_mixed_layer = calc_heat_capacity(geography,radiative_cooling_feedback) # TODO: Remove unused variables
-
+  
     co_albedo = 1.0 .- albedo
     
     ecc = eccentricity(1950)
     ob = obliquity(1950)
     per = perihelion(1950)
-    solar_forcing = calc_solar_forcing(co_albedo, ecc=ecc, ob=ob, per=per) #TODO: Add arguments [nx, ny, num_steps_year]
+     
+    
+    lambda,solar = calc_solar_forcing_params(nlatitude=ny, ntimesteps=num_steps_year, ecc=ecc, ob=ob, per=per)
+    solar_forcing = calc_solar_forcing(co_albedo, solar,nx,ny,num_steps_year)
   
     
-    return Model(diffusion_coeff, heat_capacity, albedo, solar_forcing, radiative_cooling_co2, radiative_cooling_feedback,co_albedo,compute_albedo)
-end
-
-function set_co2_concentration!(model, co2_concentration)
+    return Model(diffusion_coeff, heat_capacity, albedo, solar_forcing, radiative_cooling_co2, radiative_cooling_feedback,co_albedo,compute_albedo,geography,solar, compute_sea_ice_extent,year)
+  end
+  
+  function set_co2_concentration!(model, co2_concentration)
     model.radiative_cooling_co2 = calc_radiative_cooling_co2(co2_concentration)
-end
-
-
+  end
+  
+  
+  # Gives the possibility to interactively decide which parameters should be updated in a simulation
 
-Base.size(model::Model) = size(model.heat_capacity)
+  function update_model(model, mesh, time_step, update_geography=true, update_albedo=true, update_diffusion=true,update_heat_capacity=true, update_solar_forcing=true)
+  @unpack nx,ny = mesh
+  
+  if update_geography == true
+    set_geography!(model,mesh, time_step)
+  end
+  
+  if update_albedo == true
+    set_albedo!(model,mesh)
+    set_co_albedo!(model)
+  end
+  
+  if update_diffusion == true
+    set_diffusion_coeff!(model, mesh)
+  end
+  if update_heat_capacity == true
+    set_heat_capacity!(model)
+  end
+  if update_solar_forcing == true
+    set_solar_forcing!(model, mesh)
+  end
+  end
+  
+  
+  function set_geography!(model, mesh, time_step)
+  @unpack nx,ny = mesh
+
+  # monthly geography can be updated by reading the input maps or by computing the sea ice extent
+  # first time step is the last week of March as the simulation starts at the vernal equinox
+   if model.compute_sea_ice_extent == false
+      if time_step in (1,46,47,48)
```

Comment:
No, it does not work for other NT. I am checking these steps because the first step should be the last week of March and the last 3 steps should be the first three weeks of March, so the map for March needs to be read. Does the discretization also work for other NT? If I am not mistaken it was said in the papers that only 48 time step are possible.

Reply:
> I am checking these steps because the first step should be the last week of March and the last 3 steps should be the first three weeks of March, so the map for March needs to be read.

Ok, that makes a lot of sense. 
> Does the discretization also work for other NT? If I am not mistaken it was said in the papers that only 48 time step are possible.

The entire `numerics.jl` module works for any `NT`. However, I just tried to run the examples with `NT=96`, and noticed that there is indeed a small problem in `model.jl`, which does not allow to change `NT`. Then it's OK. You can leave the code as it is, but please add a comment (or TODO note) specifying that these routines only work for `NT=48`. We will solve that problem in the future.

Reply:
I just created an issue for this problem (#22). Please add the TODO notes in all places that you consider necessary, such that we can easily correct the problem when solving issue #22.

Reply:
OK, thank you. I will add the necessary comments.